File: src/sbml/test/TestUnitKind.c
/**
 * \file    TestUnitKind.h
 * \brief   UnitKind enumeration unit tests
 * \author  Ben Bornstein
 *
 * $Id$
 * $Source$
 */
/* Copyright 2002 California Institute of Technology and
 * Japan Science and Technology Corporation.
 *
 * This library is free software; you can redistribute it and/or modify it
 * under the terms of the GNU Lesser General Public License as published
 * by the Free Software Foundation; either version 2.1 of the License, or
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF
 * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE.  The software and
 * documentation provided hereunder is on an "as is" basis, and the
 * California Institute of Technology and Japan Science and Technology
 * Corporation have no obligations to provide maintenance, support,
 * updates, enhancements or modifications.  In no event shall the
 * California Institute of Technology or the Japan Science and Technology
 * Corporation be liable to any party for direct, indirect, special,
 * incidental or consequential damages, including lost profits, arising
 * out of the use of this software and its documentation, even if the
 * California Institute of Technology and/or Japan Science and Technology
 * Corporation have been advised of the possibility of such damage.  See
 * the GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this library; if not, write to the Free Software Foundation,
 * Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.
 *
 * The original code contained here was initially developed by:
 *
 *     Ben Bornstein
 *     The Systems Biology Markup Language Development Group
 *     ERATO Kitano Symbiotic Systems Project
 *     Control and Dynamical Systems, MC 107-81
 *     California Institute of Technology
 *     Pasadena, CA, 91125, USA
 *
 *     http://www.cds.caltech.edu/erato
 *     mailto:sbml-team@caltech.edu
 *
 * Contributor(s):
 */


#include "common/common.h"
#include "UnitKind.h"

#include <check.h>


START_TEST (test_UnitKind_equals)
{
  fail_unless( UnitKind_equals( UNIT_KIND_AMPERE , UNIT_KIND_AMPERE  ), NULL );
  fail_unless( UnitKind_equals( UNIT_KIND_INVALID, UNIT_KIND_INVALID ), NULL );
  fail_unless( UnitKind_equals( UNIT_KIND_LITER  , UNIT_KIND_LITER   ), NULL );
  fail_unless( UnitKind_equals( UNIT_KIND_LITRE  , UNIT_KIND_LITRE   ), NULL );
  fail_unless( UnitKind_equals( UNIT_KIND_METER  , UNIT_KIND_METER   ), NULL );
  fail_unless( UnitKind_equals( UNIT_KIND_METRE  , UNIT_KIND_METRE   ), NULL );

  fail_unless( UnitKind_equals(UNIT_KIND_LITER, UNIT_KIND_LITRE), NULL );
  fail_unless( UnitKind_equals(UNIT_KIND_LITRE, UNIT_KIND_LITER), NULL );

  fail_unless( UnitKind_equals(UNIT_KIND_METER, UNIT_KIND_METRE), NULL );
  fail_unless( UnitKind_equals(UNIT_KIND_METRE, UNIT_KIND_METER), NULL );

  fail_unless( !UnitKind_equals(UNIT_KIND_AMPERE, UNIT_KIND_WEBER), NULL );
}
END_TEST


START_TEST (test_UnitKind_forName)
{
  fail_unless( UnitKind_forName("ampere")    == UNIT_KIND_AMPERE   , NULL );
  fail_unless( UnitKind_forName("becquerel") == UNIT_KIND_BECQUEREL, NULL );
  fail_unless( UnitKind_forName("candela")   == UNIT_KIND_CANDELA  , NULL );
  fail_unless( UnitKind_forName("Celsius")   == UNIT_KIND_CELSIUS  , NULL );
  fail_unless( UnitKind_forName("coulomb")   == UNIT_KIND_COULOMB  , NULL );

  fail_unless( UnitKind_forName("dimensionless") == UNIT_KIND_DIMENSIONLESS,
               NULL);

  fail_unless( UnitKind_forName("farad")     == UNIT_KIND_FARAD    , NULL );
  fail_unless( UnitKind_forName("gram")      == UNIT_KIND_GRAM     , NULL );
  fail_unless( UnitKind_forName("gray")      == UNIT_KIND_GRAY     , NULL );
  fail_unless( UnitKind_forName("henry")     == UNIT_KIND_HENRY    , NULL );
  fail_unless( UnitKind_forName("hertz")     == UNIT_KIND_HERTZ    , NULL );
  fail_unless( UnitKind_forName("item")      == UNIT_KIND_ITEM     , NULL );
  fail_unless( UnitKind_forName("joule")     == UNIT_KIND_JOULE    , NULL );
  fail_unless( UnitKind_forName("katal")     == UNIT_KIND_KATAL    , NULL );
  fail_unless( UnitKind_forName("kelvin")    == UNIT_KIND_KELVIN   , NULL );
  fail_unless( UnitKind_forName("kilogram")  == UNIT_KIND_KILOGRAM , NULL );
  fail_unless( UnitKind_forName("liter")     == UNIT_KIND_LITER    , NULL );
  fail_unless( UnitKind_forName("litre")     == UNIT_KIND_LITRE    , NULL );
  fail_unless( UnitKind_forName("lumen")     == UNIT_KIND_LUMEN    , NULL );
  fail_unless( UnitKind_forName("lux")       == UNIT_KIND_LUX      , NULL );
  fail_unless( UnitKind_forName("meter")     == UNIT_KIND_METER    , NULL );
  fail_unless( UnitKind_forName("metre")     == UNIT_KIND_METRE    , NULL );
  fail_unless( UnitKind_forName("mole")      == UNIT_KIND_MOLE     , NULL );
  fail_unless( UnitKind_forName("newton")    == UNIT_KIND_NEWTON   , NULL );
  fail_unless( UnitKind_forName("ohm")       == UNIT_KIND_OHM      , NULL );
  fail_unless( UnitKind_forName("pascal")    == UNIT_KIND_PASCAL   , NULL );
  fail_unless( UnitKind_forName("radian")    == UNIT_KIND_RADIAN   , NULL );
  fail_unless( UnitKind_forName("second")    == UNIT_KIND_SECOND   , NULL );
  fail_unless( UnitKind_forName("siemens")   == UNIT_KIND_SIEMENS  , NULL );
  fail_unless( UnitKind_forName("sievert")   == UNIT_KIND_SIEVERT  , NULL );
  fail_unless( UnitKind_forName("steradian") == UNIT_KIND_STERADIAN, NULL );
  fail_unless( UnitKind_forName("tesla")     == UNIT_KIND_TESLA    , NULL );
  fail_unless( UnitKind_forName("volt")      == UNIT_KIND_VOLT     , NULL );
  fail_unless( UnitKind_forName("watt")      == UNIT_KIND_WATT     , NULL );
  fail_unless( UnitKind_forName("weber")     == UNIT_KIND_WEBER    , NULL );

  fail_unless( UnitKind_forName(NULL)     == UNIT_KIND_INVALID, NULL );
  fail_unless( UnitKind_forName("")       == UNIT_KIND_INVALID, NULL );
  fail_unless( UnitKind_forName("foobar") == UNIT_KIND_INVALID, NULL );
}
END_TEST


START_TEST (test_UnitKind_toString)
{
  const char* s;


  s = UnitKind_toString(UNIT_KIND_AMPERE);
  fail_unless(!strcmp(s, "ampere"), NULL);

  s = UnitKind_toString(UNIT_KIND_BECQUEREL);
  fail_unless(!strcmp(s, "becquerel"), NULL);

  s = UnitKind_toString(UNIT_KIND_CANDELA);
  fail_unless(!strcmp(s, "candela"), NULL);

  s = UnitKind_toString(UNIT_KIND_CELSIUS);
  fail_unless(!strcmp(s, "Celsius"), NULL);

  s = UnitKind_toString(UNIT_KIND_COULOMB);
  fail_unless(!strcmp(s, "coulomb"), NULL);

  s = UnitKind_toString(UNIT_KIND_DIMENSIONLESS);
  fail_unless(!strcmp(s, "dimensionless"), NULL);

  s = UnitKind_toString(UNIT_KIND_FARAD);
  fail_unless(!strcmp(s, "farad"), NULL);

  s = UnitKind_toString(UNIT_KIND_GRAM);
  fail_unless(!strcmp(s, "gram"), NULL);

  s = UnitKind_toString(UNIT_KIND_GRAY);
  fail_unless(!strcmp(s, "gray"), NULL);

  s = UnitKind_toString(UNIT_KIND_HENRY);
  fail_unless(!strcmp(s, "henry"), NULL);

  s = UnitKind_toString(UNIT_KIND_HERTZ);
  fail_unless(!strcmp(s, "hertz"), NULL);

  s = UnitKind_toString(UNIT_KIND_ITEM);
  fail_unless(!strcmp(s, "item"), NULL);

  s = UnitKind_toString(UNIT_KIND_JOULE);
  fail_unless(!strcmp(s, "joule"), NULL);

  s = UnitKind_toString(UNIT_KIND_KATAL);
  fail_unless(!strcmp(s, "katal"), NULL);

  s = UnitKind_toString(UNIT_KIND_KELVIN);
  fail_unless(!strcmp(s, "kelvin"), NULL);

  s = UnitKind_toString(UNIT_KIND_KILOGRAM);
  fail_unless(!strcmp(s, "kilogram"), NULL);

  s = UnitKind_toString(UNIT_KIND_LITER);
  fail_unless(!strcmp(s, "liter"), NULL);

  s = UnitKind_toString(UNIT_KIND_LITRE);
  fail_unless(!strcmp(s, "litre"), NULL);

  s = UnitKind_toString(UNIT_KIND_LUMEN);
  fail_unless(!strcmp(s, "lumen"), NULL);

  s = UnitKind_toString(UNIT_KIND_LUX);
  fail_unless(!strcmp(s, "lux"), NULL);

  s = UnitKind_toString(UNIT_KIND_METER);
  fail_unless(!strcmp(s, "meter"), NULL);

  s = UnitKind_toString(UNIT_KIND_METRE);
  fail_unless(!strcmp(s, "metre"), NULL);

  s = UnitKind_toString(UNIT_KIND_MOLE);
  fail_unless(!strcmp(s, "mole"), NULL);

  s = UnitKind_toString(UNIT_KIND_NEWTON);
  fail_unless(!strcmp(s, "newton"), NULL);

  s = UnitKind_toString(UNIT_KIND_OHM);
  fail_unless(!strcmp(s, "ohm"), NULL);

  s = UnitKind_toString(UNIT_KIND_PASCAL);
  fail_unless(!strcmp(s, "pascal"), NULL);

  s = UnitKind_toString(UNIT_KIND_RADIAN);
  fail_unless(!strcmp(s, "radian"), NULL);

  s = UnitKind_toString(UNIT_KIND_SECOND);
  fail_unless(!strcmp(s, "second"), NULL);

  s = UnitKind_toString(UNIT_KIND_SIEMENS);
  fail_unless(!strcmp(s, "siemens"), NULL);

  s = UnitKind_toString(UNIT_KIND_SIEVERT);
  fail_unless(!strcmp(s, "sievert"), NULL);

  s = UnitKind_toString(UNIT_KIND_STERADIAN);
  fail_unless(!strcmp(s, "steradian"), NULL);

  s = UnitKind_toString(UNIT_KIND_TESLA);
  fail_unless(!strcmp(s, "tesla"), NULL);

  s = UnitKind_toString(UNIT_KIND_VOLT);
  fail_unless(!strcmp(s, "volt"), NULL);

  s = UnitKind_toString(UNIT_KIND_WATT);
  fail_unless(!strcmp(s, "watt"), NULL);

  s = UnitKind_toString(UNIT_KIND_WEBER);
  fail_unless(!strcmp(s, "weber"), NULL);


  s = UnitKind_toString(UNIT_KIND_INVALID);
  fail_unless(!strcmp(s, "(Invalid UnitKind)"), NULL );

  s = UnitKind_toString(-1);
  fail_unless(!strcmp(s, "(Invalid UnitKind)"), NULL );

  s = UnitKind_toString(UNIT_KIND_INVALID + 1);
  fail_unless(!strcmp(s, "(Invalid UnitKind)"), NULL );
}
END_TEST


START_TEST (test_UnitKind_isValidUnitKindString)
{
  fail_unless( !UnitKind_isValidUnitKindString("fun-foam-unit for kids!"),
               NULL );

  fail_unless( UnitKind_isValidUnitKindString("litre"), NULL );
}
END_TEST


Suite *
create_suite_UnitKind (void) 
{ 
  Suite *suite = suite_create("UnitKind");
  TCase *tcase = tcase_create("UnitKind");


  tcase_add_test( tcase, test_UnitKind_equals   );
  tcase_add_test( tcase, test_UnitKind_forName  );
  tcase_add_test( tcase, test_UnitKind_toString );
  tcase_add_test( tcase, test_UnitKind_isValidUnitKindString );

  suite_add_tcase(suite, tcase);

  return suite;
}

File: src/bindings/ruby/test/TestUnitKind.rb
#
# This file was converted from libsbml/src/sbml/test/TestUnitKind.c
# with the help of test_c2ruby.pl (manual handling required).
#
require 'test/unit'
require 'libSBML'

class TestUnitKind < Test::Unit::TestCase

  def test_UnitKind_isValidUnitKindString
    assert_equal 0, LibSBML::UnitKind_isValidUnitKindString("fun-foam-unit for kids!",1,1)
    assert_equal 1, LibSBML::UnitKind_isValidUnitKindString("litre",2,2)
    assert_equal 0, LibSBML::UnitKind_isValidUnitKindString("liter",2,2)
    assert_equal 1, LibSBML::UnitKind_isValidUnitKindString("liter",1,2)
    assert_equal 0, LibSBML::UnitKind_isValidUnitKindString("meter",2,3)
    assert_equal 1, LibSBML::UnitKind_isValidUnitKindString("metre",2,1)
    assert_equal 1, LibSBML::UnitKind_isValidUnitKindString("meter",1,2)
    assert_equal 1, LibSBML::UnitKind_isValidUnitKindString("Celsius",2,1)
    assert_equal 0, LibSBML::UnitKind_isValidUnitKindString("Celsius",2,2)
  end

  def test_UnitKind_toString
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_AMPERE)
    assert_equal  "ampere",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_BECQUEREL)
    assert_equal  "becquerel",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_CANDELA)
    assert_equal  "candela",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_CELSIUS)
    assert_equal  "Celsius",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_COULOMB)
    assert_equal  "coulomb",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_DIMENSIONLESS)
    assert_equal  "dimensionless",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_FARAD)
    assert_equal  "farad",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_GRAM)
    assert_equal  "gram",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_GRAY)
    assert_equal  "gray",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_HENRY)
    assert_equal  "henry",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_HERTZ)
    assert_equal  "hertz",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_ITEM)
    assert_equal  "item",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_JOULE)
    assert_equal  "joule",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_KATAL)
    assert_equal  "katal",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_KELVIN)
    assert_equal  "kelvin",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_KILOGRAM)
    assert_equal  "kilogram",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_LITER)
    assert_equal  "liter",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_LITRE)
    assert_equal  "litre",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_LUMEN)
    assert_equal  "lumen",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_LUX)
    assert_equal  "lux",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_METER)
    assert_equal  "meter",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_METRE)
    assert_equal  "metre",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_MOLE)
    assert_equal  "mole",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_NEWTON)
    assert_equal  "newton",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_OHM)
    assert_equal  "ohm",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_PASCAL)
    assert_equal  "pascal",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_RADIAN)
    assert_equal  "radian",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_SECOND)
    assert_equal  "second",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_SIEMENS)
    assert_equal  "siemens",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_SIEVERT)
    assert_equal  "sievert",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_STERADIAN)
    assert_equal  "steradian",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_TESLA)
    assert_equal  "tesla",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_VOLT)
    assert_equal  "volt",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_WATT)
    assert_equal  "watt",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_WEBER)
    assert_equal  "weber",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_INVALID)
    assert_equal  "(Invalid UnitKind)",s
    s = LibSBML::UnitKind_toString(-1)
    assert_equal  "(Invalid UnitKind)",s
    s = LibSBML::UnitKind_toString(LibSBML::UNIT_KIND_INVALID + 1)
    assert_equal  "(Invalid UnitKind)",s
  end

  def test_UnitKind_forName
    assert_equal LibSBML::UNIT_KIND_AMPERE, LibSBML::UnitKind_forName("ampere")
    assert_equal LibSBML::UNIT_KIND_BECQUEREL, LibSBML::UnitKind_forName("becquerel")
    assert_equal LibSBML::UNIT_KIND_CANDELA, LibSBML::UnitKind_forName("candela")
    assert_equal LibSBML::UNIT_KIND_CELSIUS, LibSBML::UnitKind_forName("Celsius")
    assert_equal LibSBML::UNIT_KIND_COULOMB, LibSBML::UnitKind_forName("coulomb")
    assert_equal LibSBML::UNIT_KIND_DIMENSIONLESS, LibSBML::UnitKind_forName("dimensionless")
    assert_equal LibSBML::UNIT_KIND_FARAD, LibSBML::UnitKind_forName("farad")
    assert_equal LibSBML::UNIT_KIND_GRAM, LibSBML::UnitKind_forName("gram")
    assert_equal LibSBML::UNIT_KIND_GRAY, LibSBML::UnitKind_forName("gray")
    assert_equal LibSBML::UNIT_KIND_HENRY, LibSBML::UnitKind_forName("henry")
    assert_equal LibSBML::UNIT_KIND_HERTZ, LibSBML::UnitKind_forName("hertz")
    assert_equal LibSBML::UNIT_KIND_ITEM, LibSBML::UnitKind_forName("item")
    assert_equal LibSBML::UNIT_KIND_JOULE, LibSBML::UnitKind_forName("joule")
    assert_equal LibSBML::UNIT_KIND_KATAL, LibSBML::UnitKind_forName("katal")
    assert_equal LibSBML::UNIT_KIND_KELVIN, LibSBML::UnitKind_forName("kelvin")
    assert_equal LibSBML::UNIT_KIND_KILOGRAM, LibSBML::UnitKind_forName("kilogram")
    assert_equal LibSBML::UNIT_KIND_LITER, LibSBML::UnitKind_forName("liter")
    assert_equal LibSBML::UNIT_KIND_LITRE, LibSBML::UnitKind_forName("litre")
    assert_equal LibSBML::UNIT_KIND_LUMEN, LibSBML::UnitKind_forName("lumen")
    assert_equal LibSBML::UNIT_KIND_LUX, LibSBML::UnitKind_forName("lux")
    assert_equal LibSBML::UNIT_KIND_METER, LibSBML::UnitKind_forName("meter")
    assert_equal LibSBML::UNIT_KIND_METRE, LibSBML::UnitKind_forName("metre")
    assert_equal LibSBML::UNIT_KIND_MOLE, LibSBML::UnitKind_forName("mole")
    assert_equal LibSBML::UNIT_KIND_NEWTON, LibSBML::UnitKind_forName("newton")
    assert_equal LibSBML::UNIT_KIND_OHM, LibSBML::UnitKind_forName("ohm")
    assert_equal LibSBML::UNIT_KIND_PASCAL, LibSBML::UnitKind_forName("pascal")
    assert_equal LibSBML::UNIT_KIND_RADIAN, LibSBML::UnitKind_forName("radian")
    assert_equal LibSBML::UNIT_KIND_SECOND, LibSBML::UnitKind_forName("second")
    assert_equal LibSBML::UNIT_KIND_SIEMENS, LibSBML::UnitKind_forName("siemens")
    assert_equal LibSBML::UNIT_KIND_SIEVERT, LibSBML::UnitKind_forName("sievert")
    assert_equal LibSBML::UNIT_KIND_STERADIAN, LibSBML::UnitKind_forName("steradian")
    assert_equal LibSBML::UNIT_KIND_TESLA, LibSBML::UnitKind_forName("tesla")
    assert_equal LibSBML::UNIT_KIND_VOLT, LibSBML::UnitKind_forName("volt")
    assert_equal LibSBML::UNIT_KIND_WATT, LibSBML::UnitKind_forName("watt")
    assert_equal LibSBML::UNIT_KIND_WEBER, LibSBML::UnitKind_forName("weber")
    assert_equal LibSBML::UNIT_KIND_INVALID, LibSBML::UnitKind_forName(nil)
    assert_equal LibSBML::UNIT_KIND_INVALID, LibSBML::UnitKind_forName("")
    assert_equal LibSBML::UNIT_KIND_INVALID, LibSBML::UnitKind_forName("foobar")
  end

  def test_UnitKind_equals
    assert_equal 1, LibSBML::UnitKind_equals(LibSBML::UNIT_KIND_AMPERE,LibSBML::UNIT_KIND_AMPERE)
    assert_equal 1, LibSBML::UnitKind_equals(LibSBML::UNIT_KIND_INVALID,LibSBML::UNIT_KIND_INVALID)
    assert_equal 1, LibSBML::UnitKind_equals(LibSBML::UNIT_KIND_LITER,LibSBML::UNIT_KIND_LITER)
    assert_equal 1, LibSBML::UnitKind_equals(LibSBML::UNIT_KIND_LITRE,LibSBML::UNIT_KIND_LITRE)
    assert_equal 1, LibSBML::UnitKind_equals(LibSBML::UNIT_KIND_METER,LibSBML::UNIT_KIND_METER)
    assert_equal 1, LibSBML::UnitKind_equals(LibSBML::UNIT_KIND_METRE,LibSBML::UNIT_KIND_METRE)
    assert_equal 1, LibSBML::UnitKind_equals(LibSBML::UNIT_KIND_LITER,LibSBML::UNIT_KIND_LITRE)
    assert_equal 1, LibSBML::UnitKind_equals(LibSBML::UNIT_KIND_LITRE,LibSBML::UNIT_KIND_LITER)
    assert_equal 1, LibSBML::UnitKind_equals(LibSBML::UNIT_KIND_METER,LibSBML::UNIT_KIND_METRE)
    assert_equal 1, LibSBML::UnitKind_equals(LibSBML::UNIT_KIND_METRE,LibSBML::UNIT_KIND_METER)
    assert_equal 0, LibSBML::UnitKind_equals(LibSBML::UNIT_KIND_AMPERE,LibSBML::UNIT_KIND_WEBER)
  end

end
